tonic/transforms.py

```diff
@@ -316,6 +316,60 @@ def __call__(self, events):
             return functional.drop_event_numpy(events, ratio)
 
 
+@dataclass(frozen=True)
+class EventDownsampling:
+    """Applies EventDownsampling from the paper "Insect-inspired Spatio-temporal Downsampling of Event-based Input."
+        Allows:
+            1. Tonic's normal "naive" topographical scaling method to perform spatio-temporal event-based downsampling
+            2. Integrator based method to perform spatio-temporal event-based downsampling
+            3. Differentiator based method to perform spatio-temporal event-based downsampling
+            
+    Parameters:
+        sensor_size (Tuple): size of the sensor that was used [W,H,P]
+        target_size (Tuple): size of the desired resolution [W,H]
+        dt (float): temporal resolution of events in ms
+        downsampling_method (str): string stating downsampling method. Choose from ['naive', 'integrator', 'differentiator']
+        noise_threshold (int): set number of events in downsampled pixel required to emit spike. Zero by default.
+        differentiator_time_bins (int): number of differentiator time bins within dt. Two by default.
+        
+    Example:
+        >>> transform = tonic.transforms.EventDownsampling(sensor_size=(640,480,2), target_size=(20,15),
+                                                           downsampling_method='naive')
+        >>> transform = tonic.transforms.EventDownsampling(sensor_size=(640,480,2), target_size=(20,15), dt=0.5, 
+                                                           downsampling_method='integrator')
+        >>> transform = tonic.transforms.EventDownsampling(sensor_size=(640,480,2), target_size=(20,15), dt=0.5, 
+                                                           downsampling_method='differentiator', noise_threshold=2,
+                                                           differentiator_time_bins=3)
+    """
+    
+    sensor_size: Tuple[int, int, int]
+    target_size: Tuple[int, int]
+    downsampling_method: str
+    dt: Optional[float] = None
+    noise_threshold: Optional[int] = None
+    differentiator_time_bins: Optional[int] = None
+    
+    def __call__(self, events):
+        assert self.downsampling_method in ['naive', 'integrator', 'differentiator']
+        if self.downsampling_method == 'naive':
+            return functional.naive_downsample(
+                events=events, sensor_size=self.sensor_size, target_size=self.target_size
+                )
```

Comment:
Why should this be in here? This is essentially replicating the Downsample transform with different parameters. Maybe it would make more sense to add target_size to Downsample instead of here? I'm trying to reduce duplicated code here, because every line adds weight that needs to be maintained

Reply:
Changed Downsample to include target_size and added appropriate tests.